woltka/tests/test_align.py

```diff
@@ -185,6 +193,10 @@ def test_assign_parser(self):
         self.assertEqual(assign_parser('map'), parse_map_line)
         self.assertEqual(assign_parser('sam'), parse_sam_line)
         self.assertEqual(assign_parser('b6o'), parse_b6o_line)
+        with self.assertRaises(ValueError) as ctx:
```

Comment:
I believe this should work fine?
```suggestion
        with self.assertRaises(ValueError):
```

Reply:
I would also recommend using `assertRaisesRegex` to check the actual error message (can help cut down on getting the right type of error, but in the wrong place or with the wrong text.

Reply:
Ah I see what you did with `ctx` below. You simplify the check by using `assertRaisesRegex`.

Reply:
@gwarmstrong In my impression, `assertRaisesRegex` assumes that the error message you provided is a regex pattern, not an exact string, and the way it compares is `re.search` instead of `re.match`. So when you do `file not found!`, it could also match `hahaha file not found!`. So I usually tend to be more precise.

Reply:
If this is the type of error you are worried about you could use `^` to indicate the beginning of a string and `$` to indicate the beginning of string, as below:
```
>>> import re
>>> m = re.search('^abd(.*)f$', 'abd foo f')
>>> m.group(0)
'abd foo f'
>>> m = re.search('^abd(.*)f$', 'abd foo')
>>> m.group(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'group'
>>> m = re.search('^abd(.*)f$', 'haha abd foo f')
>>> m.group(0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AttributeError: 'NoneType' object has no attribute 'group'
```

Reply:
I know this regex trick and use it often. I just think that a more basic and more readable solution should be favored when the additional feature is not a necessity. The use of `assertRaises` and context manager is also seen in the official Python documentation:
https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaises